_posts/2022-09-04-configuring-security-context-in-pods-managed-by-strimzi.md

```diff
@@ -0,0 +1,175 @@
+The following example shows how you can configure the contexts of the Apache Kafka brokers in the `Kafka` custom resource:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  # ...
+  kafka:
+    template:
+      pod:
+        # Pod Security Context
+        securityContext:
+          runAsUser: 1000001
+          fsGroup: 0
+      kafkaContainer:
+        # Container security context of the Apache Kafka broker container
+        securityContext:
+          runAsUser: 2000
+  # ...
+```
+
+This gives you a lot of flexibility because you can configure different security context for every pod or container.
+But it also means you have to configure it on many different places.
+Depending on your configuration, your Apache Kafka cluster can have up to 5 different pods with up to 7 containers.
+And your Connect, Bridge or Mirror Maker deployments might add more.
+So you need to configure the security context for each of them.
+
+## Pod Security Providers
+
+To make this easier, Strimzi 0.31.0 introduces Pod Security Providers.
+Pod Security Providers can help you configure the security context for all Strimzi-managed Pods and containers from one place.
+You can configure the Pod Security Provider in the Cluster operator configuration using the environment variable `STRIMZI_POD_SECURITY_PROVIDER_CLASS`.
+Strimzi provides two different providers:
+* Baseline provider
+* Restricted provider
+
+These providers correspond to the _baseline_ and _restricted_ profiles as defined in the [Kubernetes Pod Security Standards](https://kubernetes.io/docs/concepts/security/pod-security-standards/).
+
+### Baseline provider
+
+The Baseline provider is the default provider which is used when you don't configure any other provider.
+If you want, you can also enable it explicitly by setting the `STRIMZI_POD_SECURITY_PROVIDER_CLASS` environment variable to `io.strimzi.plugin.security.profiles.impl.BaselinePodSecurityProvider`.
+Alternatively, you can also use the _shortcut_ `baseline`.
+The Baseline provider configures the Strimzi managed pods according to the [Kubernetes _baseline_ profile](https://kubernetes.io/docs/concepts/security/pod-security-standards/#baseline).
+It also provides backwards compatibility because it configures the pods in exactly the same way as previous Strimzi versions.
+So with this provider, there will be no change to how the Strimzi-managed pods are configured.
+
+### Restricted provider
+
+The Restricted provider can be enabled by setting the `STRIMZI_POD_SECURITY_PROVIDER_CLASS` environment variable to `io.strimzi.plugin.security.profiles.impl.RestrictedPodSecurityProvider`.
+Or you can also use the _shortcut_ keyword `restricted`:
+
+```yaml
+# ...
+env:
+  # ...
+  - name: STRIMZI_POD_SECURITY_PROVIDER_CLASS
+    value: io.strimzi.plugin.security.profiles.impl.RestrictedPodSecurityProvider
+  # ...
+```
```

Comment:
I agree we have the YAML example for the restricted provider here, but why not having the example for the baseline as well?

Reply:
I sguess because it's the default, so people are less likely to actually want to configure it themselves.